pkg/operators/limiter/limiter.go

```diff
@@ -132,15 +106,38 @@ func (l *limiterOperator) Priority() int {
 }
 
 type limiterOperatorInstance struct {
-	maxEntries map[datasource.DataSource]int
+	limitsPerDs map[string]int
 }
 
 func (l *limiterOperatorInstance) Name() string {
 	return name
 }
 
 func (l *limiterOperatorInstance) PreStart(gadgetCtx operators.GadgetContext) error {
-	for ds, maxEntries := range l.maxEntries {
+	for _, ds := range gadgetCtx.GetDataSources() {
+		var maxEntries int
+		if val, ok := l.limitsPerDs[""]; ok {
+			if ds.Type() != datasource.TypeArray {
+				continue
+			}
+			maxEntries = val
+		} else if val, ok := l.limitsPerDs[ds.Name()]; ok {
+			if ds.Type() != datasource.TypeArray {
+				return fmt.Errorf("%s can only be used on array data sources", ParamMaxEntries)
+			}
+			if val == -1 {
+				gadgetCtx.Logger().Debugf("limiter: disabled for data source %q", ds.Name())
+				continue
+			}
+			maxEntries = val
+		}
```

Comment:
Not related to this PR: should these two conditionals be swapped in case the value specific for a data source is present, it takes precedence over the global value?

Reply:
Nope. `GetIntValuesPerDataSource` doesn't allow to do that. You have to either use a default value for all of them `--max-entries=3`, or specify for each data source `--max-entries=exec:3,containers=1`.